### PR TITLE
MOB-4573 checkout snapshot tests

### PR DESCRIPTION
**Pull request by BartlomiejSopalaKarhoo**
## JIRA
[MOB-4573](https://karhoo.atlassian.net/browse/MOB-4573)

## DESCRIPTION
[Create a description of what you have changed in this branch]

## Types of changes

What types of changes does your code introduce to the project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix 
- [ ] New feature 
- [ ] Tech Debt
- [ ] Breaking change 
- [ ] Documentation Update
- [ ] Other: *define what type of change it is*

## CHECKLIST

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging code._
Please review the guidelines for contributing to this repository.

- [ ] All work relating to the ticket is complete (Code complete/Acceptance Criteria met)
- [ ] Lint and unit tests pass locally (if appropriate)
- [ ] Added tests that prove the fix is effective or that feature works (if appropriate)
- [ ] Documentation (if appropriate)

## SCREENSHOTS
[Add screenshots if any for changes you have made to the App/UI]

[MOB-4573]: https://karhoo.atlassian.net/browse/MOB-4573?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: KarhooUISDKTestUtils/Mocks/TestUtil.swift [hunk 1]
```diff
@@ -117,21 +117,60 @@ public class TestUtil: PrimitiveUtil {
                       licenseNumber: licenseNumber)
     }
 
-    public class func getRandomQuote(quoteId: String = getRandomString(),
+    public class func getRandomQuote(
+        quoteId: String = getRandomString(),
+
```
Comment (AleksanderWedrychowskiKarhoo) [inline KarhooUISDKTestUtils/Mocks/TestUtil.swift:122]: vertical whitespace

File: KarhooUISDKUITests/Checkout/CheckoutSnapshotSpec.swift [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+                    let journeyDetails = JourneyDetails.mockWithScheduledDate()
+                    viewModel = KarhooCheckoutViewModel(
+                        quote: quote,
+                        journeyDetails: journeyDetails,
+                        bookingMetadata: nil,
+                        router: MockCheckoutRouter()
+                    )
+                    viewModel.legalNoticeViewModel.shouldShowView = false
+                    sut.setupBinding(viewModel)
+                    navigationController.pushViewController(sut, animated: false)
```
Comment (AleksanderWedrychowskiKarhoo) [inline KarhooUISDKUITests/Checkout/CheckoutSnapshotSpec.swift:51]: Seems like this naviation methods should be moved to main beforeEach (lines 27-32).

File: KarhooUISDKUITests/Checkout/CheckoutSnapshotSpec.swift [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+                        type: "type"
+                    )
+                    let journeyDetails = JourneyDetails.mockWithScheduledDate()
+                    viewModel = KarhooCheckoutViewModel(
+                        quote: quote,
+                        journeyDetails: journeyDetails,
+                        bookingMetadata: nil,
+                        router: MockCheckoutRouter()
+                    )
+                    sut.setupBinding(viewModel)
```
Comment (AleksanderWedrychowskiKarhoo) [inline KarhooUISDKUITests/Checkout/CheckoutSnapshotSpec.swift:180]: Extract sut building & setup to designed method. It's called multiple times across the file.

File: KarhooUISDKUITests/Checkout/CheckoutSnapshotSpec.swift [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+                    navigationController.pushViewController(sut, animated: false)
+                }
+                
+                it("Flight number cell should be visible") {
```
[the remark this answers is not in the document]
Comment (BartlomiejSopalaKarhoo) [reply KarhooUISDKUITests/Checkout/CheckoutSnapshotSpec.swift:77]: had no idea about this wording rules. And I updated it in this ticket too, lets keep this code as good as possible :)

**Review by dianakarh — APPROVED** [review]

**Pull request merged**